pkg/utils/clusterinfo/fetcher.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterinfo
+
+import (
+	"context"
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/pingcap/log"
+)
+
+const prefix = "/topology"
+
+func FetchEtcd(ctx context.Context, etcdcli *clientv3.Client) ([]TiDB, Grafana,
+	AlertManager, error) {
+	resp, err := etcdcli.Get(ctx, prefix, clientv3.WithPrefix())
+	if err != nil {
+		// put error in ctx and return
+		return nil, Grafana{}, AlertManager{}, err
+	}
+	dbMap := make(map[uint64]*TiDB)
+	var grafana Grafana
+	var alertManager AlertManager
+	dbList := make([]TiDB, 0)
+	for _, kvs := range resp.Kvs {
+		key := string(kvs.Key)
+		keyParts := strings.Split(key, "/")
+		if keyParts[0] == "" {
```

Comment:
In what case this is true?

Reply:
A etcd key may start with "/", so the first part might be `""`.

Reply:
Then in what case this is not true?